### PR TITLE
#10453 Added options to CatalogTable.tsx to provide a way to pass opt…

### PR DESCRIPTION
**Pull request by shailahir**
…ions config to table.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -163,6 +164,8 @@ export const CatalogTable = (props: CatalogTableProps) => {
     typeColumn.hidden = !showTypeColumn;
   }
   const showPagination = rows.length > 20;
+  const optionsOverride =
```
Comment (benjdlambert) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:167]: Don't think you need to do this checking here, typescript is going to take care of making sure that this has something, and I think it's fine to spread null into an object anyways.

You should just be able to spread `options` in below.

File: plugins/catalog/api-report.md [hunk 2]
```diff
@@ -138,6 +138,8 @@ export interface CatalogTableProps {
   actions?: TableProps<CatalogTableRow>['actions'];
   // (undocumented)
   columns?: TableColumn<CatalogTableRow>[];
+  // (undocumented)
+  options?: TableProps<CatalogTableRow>['options'];
```
Comment (benjdlambert) [inline plugins/catalog/api-report.md:142]: think `tableOptions` is probably a better name here.

**Review by benjdlambert — COMMENTED** [review]
Thanks for this, just some small nits. :pray:

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx (hunk 1 above)
Comment (shailahir) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx]: Updated :)

File: plugins/catalog/api-report.md (hunk 2 above)
Comment (shailahir) [reply plugins/catalog/api-report.md]: Updated. changed to tableOptions

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks! :pray:

**Comment by suuus** [comment]
Congrats on your first PR merged @shailahir! Happy to count you as a contributor.

**Pull request merged**